app/components/money-display.tsx

```diff
@@ -24,130 +13,74 @@ const sizes = {
-  const symbolPosition = money.getSymbolPosition(formatUnit);
-  const formattedNumber = money.toLocaleString({
-    unit: formatUnit,
-    showCurrency: false,
+export function MoneyInputDisplay({
+  inputValue,
+  currency,
+  unit,
+  locale,
+  size = 'default',
+  className,
+}: MoneyInputDisplayProps) {
+  const money = new Money({ amount: inputValue, currency, unit });
+  const {
+    currencySymbol,
+    currencySymbolPosition,
+    integer,
+    numberOfDecimals,
+    decimalSeparator,
+  } = money.toLocalizedStringParts({
+    locale,
+    unit,
+    minimumFractionDigits: 'max',
   });
 
-  const baseProps: Omit<MoneyDisplayBaseProps, 'children'> = {
-    symbol,
-    symbolPosition,
-    ...props,
-  };
+  // Get decimal part of the input value
+  const inputHasDecimalPoint = decimalSeparator
+    ? inputValue.includes(decimalSeparator)
+    : false;
+  const inputDecimals = inputHasDecimalPoint
+    ? inputValue.split(decimalSeparator)[1]
+    : '';
 
-  if (simpleFormat) {
-    return (
-      <MoneyDisplayBase {...baseProps}>{formattedNumber}</MoneyDisplayBase>
-    );
-  }
+  // If decimal part exists in the input value, pad with zeros to numberOfDecimals places
+  const needsPaddedZeros =
+    inputHasDecimalPoint && inputDecimals.length < numberOfDecimals;
+  const paddedZeros = needsPaddedZeros
+    ? '0'.repeat(numberOfDecimals - inputDecimals.length)
+    : '';
 
-  const [moneyWhole, moneyDecimal = ''] = formattedNumber.split('.');
-  let formattedDecimal: ReactNode | null = null;
-
-  if (decimalPlaces !== undefined) {
-    // If decimalPlaces is 0, it means user has only entered a decimal point after the whole number (ie. '5.'),
-    // so we want to overwrite the entire decimal part; otherwise, overwrite the remaining decimal places
-    const overwriteZerosCount =
-      decimalPlaces === 0
-        ? moneyDecimal.length
-        : moneyDecimal.length - decimalPlaces;
-
-    // Validate that we're not overwriting non-zero digits
-    // This is to prevent users from entering a number like 1.23 and then overwriting the 2 and 3
-    if (overwriteZerosCount > 0) {
-      const digitsBeingOverwritten = moneyDecimal.slice(-overwriteZerosCount);
-      if (digitsBeingOverwritten.split('').some((d) => d !== '0')) {
-        throw new Error(
-          `Cannot overwrite non-zero digits. decimal is ${moneyDecimal}, overwriteZerosCount is ${overwriteZerosCount}`,
-        );
-      }
-    }
+  const symbol = (
+    <span className={cn('font-bold', sizes[size].symbol)}>
+      {currencySymbol}
+    </span>
+  );
 
-    // Format the decimal part with greyed zeros if needed
-    formattedDecimal = moneyDecimal && (
-      <span>
-        .{moneyDecimal.slice(0, moneyDecimal.length - overwriteZerosCount)}
-        {overwriteZerosCount > 0 && (
-          <span className="text-gray-400">
-            {'0'.repeat(overwriteZerosCount)}
-          </span>
+  return (
+    <div className={cn('inline-flex w-fit items-center', className)}>
```

Comment:
btw can this container be span as well and then we could maybe get rid of these tailwind classes because span is inline by default?